zerver/views/users.py

```diff
@@ -112,9 +113,22 @@ def update_user_backend(request: HttpRequest, user_profile: UserProfile, user_id
         # is on behalf of the administrator.
         check_change_full_name(target, full_name, user_profile)
 
+    target_user = UserProfile.objects.get(id=user_id)
+
+    clean_profile_data = []
+
     if profile_data is not None:
-        validate_user_custom_profile_data(target.realm.id, profile_data)
-        do_update_user_custom_profile_data(target, profile_data)
+        for entry in profile_data:
+            if not entry["value"]:
+                field_id = entry["id"]
+                try:
+                    do_remove_custom_profile_field_value(target_user, field_id)
+                except CustomProfileFieldValue.DoesNotExist:
```

Comment:
Why doesn't this handle the `CustomProfileField.DoesNotExist` exception?

I think it might be cleaner to just have the whole block inside `remove_user_custom_profile_data` be a `check_remove_custom_profile_field_value` function, and then have this just directly call that, rather than semi-duplicating it.

Reply:
> I think it might be cleaner to just have the whole block inside remove_user_custom_profile_data be a check_remove_custom_profile_field_value function, and then have this just directly call that

You have a point. I updated my code. I also fixed the "Why doesn't this handle the CustomProfileField.DoesNotExist exception?" issue. I admittedly forgot one edge case in my tests. But I added it now.